storage/rocksdb/ha_rocksdb.cc

```diff
@@ -2790,9 +2791,10 @@ class Rdb_trx_info_aggregator : public Rdb_tx_list_walker {
       }
 
       std::string query_str;
-      if (thd->query().str != nullptr) {
-        query_str = std::string(thd->query().str);
-      }
+      query_str.reserve(trx_query_max_len + 1);
+      size_t query_len = thd_query_safe(
+          thd, const_cast<char *>(query_str.c_str()), trx_query_max_len);
```

Comment:
const_casting const away and writing to the result of std::string::c_str is undefined behaviour - replace it with &query_str[0]? Shorter, no need for const_cast, no UB